docs/developer/make_verify.md

```diff
@@ -0,0 +1,235 @@
+
+## 2.Verify golang
+The sub-target **`verify-golang`** will run its corrsponding shell script
+
+```shell
+verify-golang:
+    hack\verify-golang.sh
+```
+
+This script will execute the golang.sh script from the **`hack/lib/golang.sh`** in current shell, 
+and verify whether Go environment is correctly set up for the kubeedge project.
+
+```shell
+kubeedge::golang::verify_golang_version
+```
+
+## 3.Verify vendor
+The sub-target **`verify-vendor`** will run its corrsponding shell script
+
+```shell
+verify-vendor:
+    hack\verify-vendor.sh
+```
+
+It is designed to verify the state of the vendor directory and related files (go.mode and go.sum)
+in kubeedge project.  
+It runs a script to update the vendor dependencies and then checks if there are 
+any uncommmited changes. 
+
+### Verify vendor and update vendor steps
+**3.1**. Setting shell options and Defining the Root Directory
+
+```shell
+set -o errexit
+set -o nounset
+set -o pipefail
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+```
+This set the **KUBEEDGE_ROOT** to the absolute path of the root directory of the project.
+
+**3.2**. Exectueing `hack/update-vendor.sh` to update the vendor dependencies.
+
+```shell
+${KUBEEDGE_ROOT}/hack/update-vendor.sh
+```
+and
+
+```shell
+for repo in $(kubeedge::util::list_staging_repos); do
+    go mod tidy    #clean up  the go.mode file
+    go mod vendor  #create a vendor directory with all dependencies.
+done
+```
+
+It will update go.mod and go.sum for each repository in the staging repos.  
+And then it will re-create a symlink in the vendor directoyr pointing to the corresponding staging respository
+
+```shell
+for repo in $(kubeedge::util::list_staging_repos); do
+  rm -fr "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
+  ln -s "../../../staging/src/github.com/kubeedge/${repo}/" "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
+done
+```
+
+From the above operations, the **`verify-vendor.sh`** and **`update-vendor.sh`** will help other packages and tools updated,
+and maintain the consistency and correctness in the project's dependencies.
+
+**3.3**. Check Status
+
+After the **`update-vendor.sh`** executed, **`check_status`** will check git status.  
+    &emsp;If result is 0, it means there are no uncommitted changes, and vendor updated successfully.  
+    &emsp;If result is not 0, it means there are still uncommitted changes, and vendor verify failed.
+
+```shell
+ret=$(kubeedge::git::check_status)
+```
+
+ 
+## 4.Verify codegen
+
+The sub-target **`verify-codegen`** will run its corrsponding shell script
+
+```shell
+verify-codegen:
+    hack\verify-codegen.sh
+```
+
+### Verify codegen steps
+
+**4.1**. Defining Variables and clean up environment
+
+```shell
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+DIFFROOT="${KUBEEDGE_ROOT}/pkg"
+TMP_DIFFROOT="${KUBEEDGE_ROOT}/_tmp/pkg"
+_tmp="${KUBEEDGE_ROOT}/_tmp"
+cleanup
```

Comment:
Now kubeedge has migrated the automatically generated code from the pkg directory to the staging directory. You can view the latest verify-codegen script. https://github.com/kubeedge/kubeedge/blob/master/hack/verify-codegen.sh#L23

Reply:
Updated latest DIFFROOT and TMP_DIFFROOT